### PR TITLE
stress test impactless firmware update

### PR DESCRIPTION
**Pull request by srinathshiv**

### CODE REVIEW

File: runtime/tests/runtime_integration_tests/integration_tests.rs [hunk 1]
```diff
@@ -88,6 +88,49 @@ fn test_update() {
...
+
+    let stress_num: u32 = 500;
+
+    for _ in 0..stress_num{
+        model
+        .step_until(
+        |m|m.soc_mbox().status().read().mbox_fsm_ps().mbox_idle());
+
+        model
+        .mailbox_execute( u32::from(CommandId::FIRMWARE_LOAD), &image)
```
Comment (bluegate010) [inline runtime/tests/runtime_integration_tests/integration_tests.rs:119]: Might be interesting to ping-pong between multiple versions.

**Review by bluegate010 — APPROVED** [review]

**Comment by sree-revoori1** [comment]
Can you delete "Stress test: Perform many hitless updates in a row | N/A | N/A" from the test gaps section of ROM's test-coverage.md?

**Comment by sree-revoori1** [comment]
I think running it 500 times would take too long. Can you look into running this test only in nightly runs?

**Comment by korran** [comment]
> I think running it 500 times would take too long. Can you look into running this test only in nightly runs?

Consider adding a slow_tests feature to this crate, similar to https://github.com/chipsalliance/caliptra-sw/blob/5911bb4f298459898cab4503b66646bc5d3d4da9/builder/bin/image_gen.rs#L80

Then we need to add a nightly workflow that enables the slow tests...

**Comment by sree-revoori1** [comment]
Could you add this test to runtime's test-coverage.md?

File: runtime/tests/runtime_integration_tests/integration_tests.rs (hunk 1 above)
Comment (srinathshiv) [reply runtime/tests/runtime_integration_tests/integration_tests.rs:119]: Thanks for your insight, will update the test to handle multiple versions.

**Comment by srinathshiv** [comment]
Hello all, the request is in a good state. Are there any changes that I should be making?

**Review by mhatrevi — APPROVED** [review]

File: runtime/tests/runtime_integration_tests/integration_tests.rs [hunk 2]
```diff
@@ -88,6 +88,72 @@ fn test_update() {
...
+            model
+                .step_until_output_contains("Caliptra RT listening for mailbox commands...")
+                .unwrap();
+
+            //Check if the new firmware is actually the one we built
+            let fw_rev = model.soc_ifc().cptra_fw_rev_id().read();
+            assert_eq!(fw_rev[0], 0xaaaaaaaa);
+            assert_eq!(fw_rev[1], app_versions[image_select]);
+            image_select -= 1;
+        }
```
Comment (jhand2) [inline runtime/tests/runtime_integration_tests/integration_tests.rs:153]: You should be able to simplify this quite a bit

```rust
if image_select == 0 {
    image_select = 1;
} else {
    image_select = 0;
}

model
  .mailbox_execute(u32::from(CommandId::FIRMWARE_LOAD), &image[image_select])
  .unwrap();

model
  .step_until_output_contains("Caliptra RT listening for mailbox commands...")
  .unwrap();

//Check if the new firmware is actually the one we built
let fw_rev = model.soc_ifc().cptra_fw_rev_id().read();
assert_eq!(fw_rev[0], 0xaaaaaaaa);
assert_eq!(fw_rev[1], app_versions[image_select]);
```

File: runtime/tests/runtime_integration_tests/integration_tests.rs [hunk 3]
```diff
@@ -88,6 +88,72 @@ fn test_update() {
...
+            .to_bytes()
+            .unwrap(),
+    ];
+    let mut model = run_rt_test(None, None, None);
+
+    let stress_num: u32 = 500;
+    let mut image_select = 0;
+
+    for _ in 0..stress_num {
+        model.step_until(|m| m.soc_mbox().status().read().mbox_fsm_ps().mbox_idle());
```
Comment (jhand2) [inline runtime/tests/runtime_integration_tests/integration_tests.rs:123]: Should be able to do this just once before entering the loop

File: runtime/tests/runtime_integration_tests/integration_tests.rs (hunk 2 above)
Comment (srinathshiv) [reply runtime/tests/runtime_integration_tests/integration_tests.rs]: Looks good. Updated the code.

File: hw-latest/caliptra-rtl
Comment (mhatrevi) [inline hw-latest/caliptra-rtl:1]: Can you revert the change to updating the RTL module?

**Review by mhatrevi — CHANGES_REQUESTED** [review]
Can you revert the change to updating the RTL module?

File: hw-latest/caliptra-rtl
Comment (srinathshiv) [reply hw-latest/caliptra-rtl:1]: Is there a specific commit I should revert the RTL submodule to?

**Comment by srinathshiv** [comment]
Previously I reverted a merge commit which caused multiple files to be reverted back. As a result, number of files got changed to 141 and that messed up the PR. So, I was asked to fix it. 

I tried by resetting the repo to an old commit (5548cc45, oct 27). I did a hard reset because that serves my purpose for this scenario. That fixed the pull request. Then I merged the upstream(caliptra-sw) into the forked repo to make it up to date and added my changes on top of it based on the previous comments. Now, the pull request is clean and open to comments.

**Review by mhatrevi — DISMISSED** [review]

File: rom/dev/doc/test-coverage/test-coverage.md [hunk 4]
```diff
@@ -138,7 +138,7 @@ Expand `smoke_test` to perform a hitless update and confirm everything is mixed
 Validate fix for #817: warm reset during hitless update | N/A | N/A
 Validate fix for #628: warm reset during cold reset | N/A | N/A
 Add test for CPU fault, and that extended error info is populated correctly | N/A | N/A
-Stress test: Perform many hitless updates in a row | N/A | N/A
+Stress test: Boot caliptra 1000 times with a different UDS identity each time, and confirm generated certs are valid. This should expose x509 serialization bugs. | N/A | N/A
```
[the remark this answers is not in the document]
Comment (srinathshiv) [reply rom/dev/doc/test-coverage/test-coverage.md:141]: Hi @sree-revoori1 Do you mean the "Stress test: Boot caliptra...." line? I think it has to be done in a seperate pull request. It's coming from the main branch(of caliptra-sw).

[the remark this answers is not in the document]
Comment (sree-revoori1) [reply rom/dev/doc/test-coverage/test-coverage.md:141]: Yes, but it has been done already in this PR - https://github.com/chipsalliance/caliptra-sw/pull/1017 , so we should remove it.

[the remark this answers is not in the document]
Comment (srinathshiv) [reply rom/dev/doc/test-coverage/test-coverage.md:141]: OK, I'll be removing that line on behalf of 1017.

File: rom/dev/doc/test-coverage/test-coverage.md [hunk 5]
```diff
@@ -136,6 +136,9 @@ Ensure that boot ROM can load a 128k bundle into ICCM (assert ICCM contents in t
 Test Scenario| Test Name | ROM Error Code
 ---|---|---
 Expand `smoke_test` to perform a hitless update and confirm everything is mixed into the identity correctly. | N/A | N/A
-Stress test: Perform many hitless updates in a row | N/A | N/A
+Validate fix for #817: warm reset during hitless update | N/A | N/A
```
Comment (sree-revoori1) [inline rom/dev/doc/test-coverage/test-coverage.md:139]: Lines 139-142 should be deleted here.

Comment (sree-revoori1) [reply rom/dev/doc/test-coverage/test-coverage.md]: I see line 140 is still there? There should be no new additions to rom/dev/doc/test-coverage/test-coverage.md.

Comment (srinathshiv) [reply rom/dev/doc/test-coverage/test-coverage.md]: I wanted to check with you on that, thanks for pointing it out. Removed the line in the latest commit.

File: runtime/tests/runtime_integration_tests/test_boot.rs [hunk 6]
```diff
@@ -80,6 +80,60 @@ fn test_update() {
...
+            .unwrap(),
+        caliptra_builder::build_and_sign_image(&FMC_WITH_UART, &APP_WITH_UART, image_options_1)
+            .unwrap()
+            .to_bytes()
+            .unwrap(),
+    ];
+
+    let mut model = run_rt_test(None, None, None);
+
+    let stress_num: u32 = 500;
```
Comment (jhand2) [inline runtime/tests/runtime_integration_tests/test_boot.rs:110]: Instead of fully disabling the tests based on feature, can you do something like https://github.com/chipsalliance/caliptra-sw/blob/4cc5df2e6a286200687843d5d88253ba54efb800/rom/dev/tests/rom_integration_tests/test_idevid_derivation.rs#L82-L86

This makes sure the test gets run in the presubmits too instead of just nightly to prevent breakages.

Comment (srinathshiv) [reply runtime/tests/runtime_integration_tests/test_boot.rs]: Updated the test to be run atleast once when the feature is disabled.

**Review by mhatrevi — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]
